test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionMetadataProviderTest.cs

```diff
@@ -148,6 +148,29 @@ public void FromStringExpression_SetsContainerAsExpected()
             Assert.Same(myModel, metadata.Container.Model);
         }
 
+        // This is a special case that's basically only possible with Razor Pages.
```

Comment:
@rynowak discussed offline and he's going to try a few more cases e.g. fields, `static` properties, and visibility-challenged properties. Indexers are handled elsewhere. Method groups probably can't be referenced in a Linq expression but @rynowak will try.